app/factories/shopify/product_factory.rb

```diff
@@ -0,0 +1,59 @@
+module Shopify
+  class ProductFactory
+    def initialize(spree_product, shopify_product, logger = nil)
+      @logger = logger || default_logger
+      @spree_product = spree_product
+      @shopify_product = shopify_product
+    end
+
+    def perform
```

Comment:
Instead of having to set the shopify_product, why not just return a hash of values instead.